### PR TITLE
Initial dex-ui project setup

### PR DESCRIPTION
**Pull request by marshall-bean**
Initial dex-ui project structure, with typescript, eslint, prettier, vite, etc.

Used sensible defaults which we can tweak if needed (e.g. different eslint rules, etc) and could even adopt some or all from the main UI project if we want.

I also added a simple GH action to build the project in lieu of tests, just to give us some extra confidence.

### CODE REVIEW

File: projects/dex-ui/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (0xalecks) [inline projects/dex-ui/.prettierrc:1]: remove file

File: projects/dex-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "dex-ui",
+  "description": "A user interface for Beanstalk Dex",
+  "private": true,
+  "version": "1.0.0",
+  "license": "MIT",
+  "scripts": {
+    "start": "vite --port 2424",
+    "build": "vite build",
+    "lint": "npx eslint src",
```
Comment (0xalecks) [inline projects/dex-ui/package.json:10]: why run through npx, these will be installed locally anyway

File: projects/dex-ui/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "lint:fix": "npm run lint -- --fix",
+    "prettier": "npx prettier src --check",
+    "prettier:fix": "npm run prettier -- --write",
+    "format": "npm run prettier:fix && npm run lint:fix"
+  },
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
+  },
+  "devDependencies": {
```
Comment (0xalecks) [inline projects/dex-ui/package.json:20]: add lint-staged and a .lintstagedrc (or whatever). see the other projects for reference

File: projects/dex-ui/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (0xalecks) [inline projects/dex-ui/tsconfig.json:1]: Should extend root tsconfig and overwrite only what's needed

File: projects/dex-ui/package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "^7.32.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "prettier": "^2.8.4",
+    "typescript": "^4.9.5",
+    "vite": "^4.1.4",
+    "vite-tsconfig-paths": "^4.0.5"
+  }
+}
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/dex-ui/package.json:38]: Coming back here after finishing the review to add another comment. The yarn.lock file had a lot of new entries which meant that we are adding new things. I would've expected this project, at this stage, to really not have to add any new dependencies and just mirror the UI. 

Comparing the deps here with the existing UI, this project is using 
- newer versions of things (ex vite)
- things that UI doesn't use (ex prettier, typescript

Think we should just use the same versions for these things to keep things consistent and avoid multiple versions of the same stuff

**Review by 0xalecks — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -37,6 +37,8 @@
     "ui:build": "yarn workspace ui build",
     "ui:test": "yarn workspace ui test",
     "test:browser": "yarn workspace tests test:browser",
+    "dex-ui:start": "yarn workspace dex-ui start",
+    "dex-ui:build": "yarn workspace dex-ui build",
```
[the remark this answers is not in the document]
Comment (marshall-bean) [reply package.json:41]: OK will remove these

File: projects/dex-ui/.eslintrc [hunk 7]
```diff
@@ -0,0 +1,49 @@
+{
```
[the remark this answers is not in the document]
Comment (marshall-bean) [reply projects/dex-ui/.eslintrc:1]: I don't see any `.eslintrc` in root, can you point me to it?

File: projects/dex-ui/package.json (hunk 2 above)
Comment (marshall-bean) [reply projects/dex-ui/package.json:10]: Will fix

File: projects/dex-ui/package.json (hunk 3 above)
Comment (marshall-bean) [reply projects/dex-ui/package.json:20]: I assume you mean with husky, will add.

File: projects/dex-ui/.prettierignore [hunk 8]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
[the remark this answers is not in the document]
Comment (marshall-bean) [reply projects/dex-ui/.prettierignore:1]: 👍

File: projects/dex-ui/tsconfig.json (hunk 4 above)
Comment (marshall-bean) [reply projects/dex-ui/tsconfig.json:1]: 👍

File: projects/dex-ui/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (marshall-bean) [reply projects/dex-ui/package.json:39]: I'll align the versions between dex-ui and ui.

File: projects/dex-ui/package.json (hunk 3 above)
Comment (marshall-bean) [reply projects/dex-ui/package.json:21]: I realized husky and the lint-staged pre-commit hook isn't working even in the UI project. We can split into a separate PR against master if we wanted to.

**Pull request merged**